src/cpp/r/RCntxt.cpp

```diff
@@ -192,7 +192,8 @@ bool RCntxt::operator==(const RCntxt& other) const
 
    // Also equivalent if they refer to the same call at the same stack position and have the same
    // source references
-   if (other.call() == call() && 
+   if (!other.isNull() &&
```

Comment:
For symmetry, would it make sense to test `other.isNull() == isNull()` here?

If not, could we move this check earlier just to sort of indicate the check here is "special" compared to the other members being checked?

(If I understand correctly, the issue here is that we had a non-null context that "looked like" a null context, because all of call / evaldepth / srcref had their "default" values... is that right?)

Reply:
Yeah, that seemed to be what was going on. I think I probably could change it to compare isNull to isNull and it would have the same effect, it didn't seem likely to me that they could both be null, otherwise the function would have returned true on the direct comparison in the first if block, but for the sake of readability I'm fine with spelling it out here